Legifrance.js

```diff
@@ -40,252 +40,261 @@ Thanks to Sebastian Karcher and Aurimas Vinckevicius
+	
+	var e;// Cours administratives d'appel avec publication // très rares cas sans publication
+	e = title.match(/(Cour administrative .*), (.*), (s*[0-9/]+), (.*), (.*Lebon)/);
+	if (e) {
+		cour = e[1];
+		formation = e[2];
+		date = e[3];
+		numero = e[4];
+		publication = e[5];
+		newItem.court = cour;
+		newItem.extra = formation;
+		newItem.date = date;
+		newItem.docketNumber = numero;
+		newItem.reporter = publication;
+	}
+	
+	var f; // tribunaux administratifs avec chambre
+	f = title.match(/(|Tribunal Administratif|administratif.*), (.*chambre), (s*[0-9/]+), (s*[0-9]+)/);
+	if (f) {
+		cour = f[1];
+		formation = f[2];
+		date = f[3];
+		numero = f[4];
+		newItem.court = 'Tribunal ' + cour;
+		newItem.date = date;
+		newItem.docketNumber = numero;
+	}
+	
+	var g; // tribunaux administratifs sans chambre avec publication
+	g = title.match(/(Tribunal Administratif|administratif.*), du (.*), (s*[0-9-]+), (.*Lebon)/);
+	if (g) {
+		cour = g[1];
+		date = g[2];
+		numero = g[3];
+		publication = g[4];
+		newItem.court = 'Tribunal ' + cour;
+		newItem.date = date;
+		newItem.docketNumber = numero;
+		newItem.reporter = publication;
+	}
+	
+	// Note : présence d'autres cas pour les TA
+	
+	var h; // Cour de cassation
+	h = title.match(/(Cour de cassation), (.*), (.*), (s*[0-9-. ]+), (.*)/);
+	if (h) {
+		nature = h[1];
+		formation = h[2];
+		date = h[3];
+		numero = h[4];
+		publication = h[5];
+		newItem.court = 'Cour de cassation';
+		if (nature) newItem.tags.push(nature);
+		newItem.extra = formation;
+		newItem.date = date;
+		newItem.docketNumber = numero;
+		newItem.reporter = publication;
+	}
+	
+	var i; // cours d'appel et tribunaux
+	i = title.match(/(Cour d'appel.*|Tribunal.*|Conseil.*|Chambre.*|Juridiction.*|Commission.*|Cour d'assises.*) de (.*), (.*), (s*[0-9/]+)/);
+	if (i) {
+		cour = i[1];
+		var lieu = i[2];
+		date = i[3];
+		numero = i[4];
+		newItem.court = cour + ' de ' + lieu;
+		newItem.date = date;
+		newItem.docketNumber = numero;
+	}
+	
+	var j;// Tribunal des conflits - Base CASS
+	j = title.match(/(Tribunal des conflits), (.*), (.*), (s*[0-9-. ]+), (.*)/);
+	if (j) {
+		nature = j[2];
+		date = j[3];
+		numero = j[4];
+		publication = j[5];
+		newItem.court = 'Tribunal des conflits';
+		if (nature) newItem.tags.push(nature);
+		newItem.date = date;
+		newItem.docketNumber = numero;
+		newItem.reporter = publication;
+	}
+	newItem.complete();
+}
 
-		}
 
-		var e; // CNIL
-		e = title.match(/(Délibération) (s*[0-9-]+) du ((s*[0-9]+) (.*) (s*[0-9]+))/)
-		if (e) {
-			var nameOfAct = e[1];
-			var code = e[2];
-			var date = e[3];
-			newItem.nameOfAct = nameOfAct + ' de la Commission Nationale de l\'Informatique et des Libertés';
-			newItem.code = code;
-			newItem.date = date;
+function scrapelegislation(doc) { //Législation
+	var code, date, UnParam;
+	var newItem = new Zotero.Item("statute");
+	
+	var title = ZU.xpathText(doc, '//h2[@class="title"]');
+	newItem.title = title;
+	var MonUrl = doc.location.href;
```

Comment:
One more question:
I may be reading this too quickly, but couldn't we reduce the entire thing below to:
`MonUrl = MonUrl.replace("jsessionid", "?").replace(/(dateTexte[^&]+|categorieLien[^&]+)\&/g, "");`
(might need some tweaking to deal with end of URL)